x-pack/plugins/security_solution/public/flyout/shared/components/copy_to_clipboard.tsx

```diff
@@ -19,67 +18,43 @@ export interface CopyToClipboardProps {
    * Function to modify the raw value before saving to the clipboard
    */
   modifier?: (rawValue: string) => string;
-  /**
-   * Button main text (next to icon)
-   */
-  text?: ReactElement;
-  /**
-   * Icon name (value coming from EUI)
-   */
-  iconType: EuiButtonEmptyProps['iconType'];
-  /**
-   * Button size (values coming from EUI)
-   */
-  size?: EuiButtonEmptyProps['size'];
-  /**
-   * Optional button color
-   */
-  color?: EuiButtonEmptyProps['color'];
-  /**
-   * Aria label value for the button
-   */
-  ariaLabel: string;
   /**
    Data test subject string for testing
    */
   ['data-test-subj']?: string;
 }
 
 /**
- * Copy to clipboard component
+ * Copy to clipboard wrapper component. It allows adding a copy to clipboard functionality to any element.
+ * It expects the value to be copied with an optional function to modify the value if necessary.
  */
 export const CopyToClipboard: FC<CopyToClipboardProps> = ({
   rawValue,
   modifier,
-  text,
-  iconType,
-  size = 'm',
-  color = 'primary',
-  ariaLabel,
   'data-test-subj': dataTestSubj,
+  children,
 }) => {
+  const copyFunction = (copy: Function) => {
```

Comment:
What do you think about making `copyFunction` a util function instead? 

```
export const copyFunction = (
  copy: Function,
  rawValue: string,
  modifier?: (rawValue: string) => string
) => {
  copy();

  if (modifier) {
    const modifiedCopyValue = modifier(rawValue);
    copyToClipboard(modifiedCopyValue);
  } else {
    copyToClipboard(rawValue);
  }
};

Reply:
implemented, thanks for the great idea!